packages/flutter/lib/src/material/bottom_sheet.dart

```diff
@@ -260,6 +269,17 @@ class _BottomSheetState extends State<BottomSheet> {
         child: widget.builder(context),
       ),
     );
+
+    if (constraints != null) {
+      bottomSheet = Align(
+        alignment: Alignment.bottomCenter,
```

Comment:
This looks like the right default and it's not clear that we should enable overriding it. The API doc for the new constraints parameter should probably mention the implicit alignment though.